cmd/openfga/main_test.go

```diff
@@ -649,6 +706,117 @@ func GRPCExpandTest(t *testing.T, tester OpenFGATester) {
 
 }
 
+// TestCheckWorkflows are tests that involve workflows that define assertions for
+// Checks against multi-model stores etc..
+func TestCheckWorkflows(t *testing.T) {
+
+	tester, err := newOpenFGATester(t)
+	require.NoError(t, err)
+	defer tester.Cleanup()
+
+	conn := connect(t, tester)
+	defer conn.Close()
+
+	client := openfgapb.NewOpenFGAServiceClient(conn)
+
+	/*
+	 * TypedWildcardsFromOtherModelsIgnored ensures that a typed wildcard introduced
+	 * from a prior model does not impact the Check outcome of a model that should not
+	 * involve it. For example,
+	 *
+	 * type user
+	 * type document
+	 *   relations
+	 *     define viewer: [user:*]
+	 *
+	 * write(document:1#viewer@user:*)
+	 *
+	 * type user
+	 * type document
+	 *	relations
+	 *	  define viewer: [user]
+	 *
+	 * check(document:1#viewer@user:jon) --> {allowed: false}
+	 */
+	t.Run("TypedWildcardsFromOtherModelsIgnored", func(t *testing.T) {
```

Comment:
What happens in this scenario?

1. Run an old version of OpenFGA without this PR
2. Write the old auth model, and a tuple involving `user:*`
3. Run OpenFGA with this PR
4. Write the new auth model
5. Call Check

Reply:
The opposite wont' work as well because of the difference in how we treat `user:*`.

For example the following workflow is problematic

```
type document
  relations
    define viewer as self

write(document:1#viewer@user:*)
check(document:1#viewer@user:*) --> {allowed: true}
check(document:1#viewer@user:jon) --> {allowed: false}

type user
type document
  relations
    define viewer: [user:*, user]

check(document:1#viewer@user:*) --> {allowed: true} // this is fine because it's a direct checkout (e.g. ReadUserTuple)
check(document:1#viewer@user:jon) --> {allowed: false} // this is problematic
```

This issue exists because the tuple written in the old version of OpenFGA will look like 
| object_type | object_id | relation | _user  | user_type |
|-------------|-----------|----------|--------|-----------|
| ...         | ...       | ...      | user:* | user      |

But under this new version of OpenFGA we'll try to read tuples of user_type `userset` and it won't find any.

Reply:
Discussed this issue with the team and we've decided to proceed forward with this incompatibility. Developers will have to migrate tuples that have user values that look like typed wildcards by removing them and re-adding them if they intended to have public access.